packages/@sanity/codegen/src/typescript/__tests__/typeGenerator.test.ts

```diff
@@ -426,3 +426,48 @@ export type AllSanitySchemaTypes = OptionalData;"
     expect(someOtherTypeOut).toMatchSnapshot()
   })
 })
+
+describe('generateQueryMap', () => {
+  test('should generate a map of query results', () => {
+    const schema: SchemaType = [
+      {
+        type: 'document',
+        name: 'author',
+        attributes: {
+          _id: {
+            type: 'objectAttribute',
+            value: {type: 'string'},
+          },
+          name: {
+            type: 'objectAttribute',
+            value: {type: 'string'},
+            optional: true,
+          },
+        },
+      },
+    ]
+
+    const queries = [
+      {
+        typeName: 'AuthorsResult',
+        query: '*[_type "author"]',
+      },
+      {
+        typeName: 'FirstAuthorResult',
+        query: '*[_type "author"][0]',
+      },
+    ]
+
+    const typeGenerator = new TypeGenerator(schema)
+    const actualOutput = typeGenerator.generateQueryMap(queries)
+
+    expect(actualOutput).toMatchInlineSnapshot(`
+"declare module \\"@sanity/client\\" {
+  interface SanityQueries {
+    \\"*[_type \\\\\\"author\\\\\\"]\\": AuthorsResult;
+    \\"*[_type \\\\\\"author\\\\\\"][0]\\": FirstAuthorResult;
+  }
+}"
+`)
+  })
```

Comment:
@sgulseth note that I opted not to `extend QueryReturn` here. In my local testing it was not necessary and overriding `SanityQueries` just works.